example/test.py

```diff
@@ -0,0 +1,469 @@
+        """
+        ps_data = list_container_processes(self.web_container)
+
+        assert_that(ps_data, HasLength(5))
+
+        assert_that(ps_data.pop(0), Equals(
+            ['1', 'root',
+             'tini -- django-entrypoint.sh mysite.wsgi:application']))
+
+        # The next process we have no control over the start order or PIDs...
+        ps_data = [data[1:] for data in ps_data]  # Ignore the PIDs
+        assert_that(ps_data, MatchesSetwise(*map(Equals, [
+            ['root', 'nginx: master process nginx -g daemon off;'],
+            ['nginx', 'nginx: worker process'],
+            ['django',
+             '/usr/local/bin/python /usr/local/bin/gunicorn '
+             'mysite.wsgi:application --pid /var/run/gunicorn/gunicorn.pid '
+             '--bind unix:/var/run/gunicorn/gunicorn.sock --umask 0117'],
+            # No obvious way to differentiate Gunicorn master from worker
+            ['django',
+             '/usr/local/bin/python /usr/local/bin/gunicorn '
+             'mysite.wsgi:application --pid /var/run/gunicorn/gunicorn.pid '
+             '--bind unix:/var/run/gunicorn/gunicorn.sock --umask 0117'],
+        ])))
+
+    def test_database_tables_created(self):
+        """
+        When the web container is running, a migration should have completed
+        and there should be some tables in the database.
+        """
+        psql_output = infra_containers['db'].exec_run(
+            ['psql', '-qtA', '--dbname', 'mysite', '-c',
+             ('SELECT COUNT(*) FROM information_schema.tables WHERE '
+              "table_schema='public';")],
+            user='postgres').decode('utf-8')
+
+        count = int(psql_output.strip())
+        assert_that(count, GreaterThan(0))
+
+    def get(self, path, **kwargs):
+        return requests.get(
+            'http://127.0.0.1:{}{}'.format(self.web_port, path), **kwargs)
+
+    def test_admin_site_live(self):
+        """
+        When we get the /admin/ path, we should receive some HTML for the
+        Django admin interface.
+        """
+        response = self.get('/admin/')
+
+        assert_that(response.headers['Content-Type'],
+                    Equals('text/html; charset=utf-8'))
+        assert_that(response.text,
+                    Contains('<title>Log in | Django site admin</title>'))
+
+    def test_nginx_access_logs(self):
+        """
+        When a request has been made to the container, Nginx logs access logs
+        to stdout
+        """
+        # Make a request to see the logs for it
+        self.get('/')
+
+        # Get the last line of stdout, which should be the access log entry for
+        # the request we just made
+        # FIXME: Sometimes for the last line we just get an empty string, so we
+        # have to try process the last 2
+        last_log_lines = (
+            self.web_container.logs(stdout=True, stderr=False, tail=2)
+            .decode('utf-8').split('\n'))
+        log_line = last_log_lines[1] or last_log_lines[0]
```

Comment:
Instead of looking at the last couple of lines and assuming they belong to us, we should explicitly skip anything that was logged before our request. Maybe have `old_log_count = len(self.web_container.logs(...).decode('utf-8').split('\n'))` before the request and then `log_lines = self.web_container.logs(...).decode('utf-8').split('\n')[old_log_count:]` here?

It's probably worth splitting the log extraction out into a helper function.